docs/decisions/0003-plugin-contexts.rst

```diff
@@ -0,0 +1,72 @@
+Plugin Contexts
+---------------
+
+Status
+======
+Draft
+
+Context
+=======
+edx-platform contains a plugin system which allows new Django apps to be installed inside the LMS and Studio without requiring the LMS/Studio to know about them. This is what enables us to move to a small and extensible core. While we had the ability to add settings, URLs, and signal handlers in our plugins, we wasn't a any way for a plugin to affect the commonly used pages that the core was delivering. Thus a plugin couldn't change any details on the dashboard, courseware, or any other rendered page that the platform delivered.
+
+Decisions
+=========
+We have added the ability to add page context additions to the plugin system. This means that a plugin will be able to add context any view where it is enabled. To support this we have decided:
+# Plugins will define a callable function that the LMS and/or studio can import and call, which will return additional context to be added.
+# Every page that a plugin wants to add context to, must add a line to add the plugin contexts directly before the render.
+# All view + plugin data will exist in the same dictionary. To better protect against dictionary key collisions, it is suggested that you prefix your new context items with your app name (e.g. return {"myapp__some_variable": True} vs {"some_variable": True})
```

Comment:
This was my original thought, but for some reason had thought the templating system would handle it nicely. I went back and checked and it seems like it will, so I'll move to this style.